pkg/middleware/filter.go

```diff
@@ -5,42 +5,43 @@ import (
 	"net/url"
 	"strings"
 
-	"github.com/gobuffalo/buffalo"
 	"github.com/gomods/athens/pkg/errors"
 	"github.com/gomods/athens/pkg/module"
 	"github.com/gomods/athens/pkg/paths"
+	"github.com/gorilla/mux"
 )
 
 // NewFilterMiddleware builds a middleware function that implements the
 // filters configured in the filter file.
-func NewFilterMiddleware(mf *module.Filter, upstreamEndpoint string) buffalo.MiddlewareFunc {
+func NewFilterMiddleware(mf *module.Filter, upstreamEndpoint string) mux.MiddlewareFunc {
 	const op errors.Op = "actions.NewFilterMiddleware"
-
-	return func(next buffalo.Handler) buffalo.Handler {
-		return func(c buffalo.Context) error {
-			mod, err := paths.GetModule(c)
-
+	return func(h http.Handler) http.Handler {
+		f := func(w http.ResponseWriter, r *http.Request) {
+			mod, err := paths.GetModule(r)
 			if err != nil {
 				// if there is no module the path we are hitting is not one related to modules, like /
-				return next(c)
+				h.ServeHTTP(w, r)
+				return
 			}
-
 			rule := mf.Rule(mod)
 			switch rule {
 			case module.Exclude:
 				// Exclude: ignore request for this module
-				return c.Render(http.StatusForbidden, nil)
+				w.WriteHeader(http.StatusForbidden)
+				return
 			case module.Include:
 				// Include: please handle this module in a usual way
-				return next(c)
+				h.ServeHTTP(w, r)
+				return
 			case module.Direct:
 				// Direct: do not store modules locally, use upstream proxy
-				newURL := redirectToUpstreamURL(upstreamEndpoint, c.Request().URL)
-				return c.Redirect(http.StatusSeeOther, newURL)
+				newURL := redirectToUpstreamURL(upstreamEndpoint, r.URL)
+				http.Redirect(w, r, newURL, http.StatusSeeOther)
```

Comment:
Can we simply do return http.Redirect and same for the above?

Reply:
@manugupt1 we cannot return `http.Redirect` because it does not return anything, nor does the http.HandlerFunc has a return type :)

Reply:
Interesting, I was not aware that about with go. Some languages return nil and I do not use go daily.